src/Validators/IntegerValueValidator.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\FunValidators\Validators;
+
+use WMDE\FunValidators\ConstraintViolation;
+use WMDE\FunValidators\ValidationResult;
+
+/**
+ * @license GNU GPL v2+
+ */
+class IntegerValueValidator {
+
+	public function validate( string $input ): ValidationResult {
```

Comment:
I would have left out the type hint on the basis that validators should be as flexible as possible and should not require type conversion before the actual validation, but make some for of type checking/casting part of the validation, instead of crashing the application with a type error in case a wrong type snuck in at runtime.

Seeing that we don't have  consistent interface for the validators anyway and that you'd have to cast to string internally, because `ctype_digit` will return `false` on numeric values, I'm fine with the type hint.